codegen/cuekind/def.cue

```diff
@@ -124,6 +124,10 @@ Kind: S={
 				// backend indicates whether back-end Go code should be generated for this kind's schema
 				backend: bool | *S.codegen.backend
 			}
+			// seledtableFields is a list of additional fields which can be used in kubernetes field selectors for this version.
+			// Fields must be from the root of the schema, i.e. 'spec.foo', and have a string type.
+			// Fields cannot include custom metadata (TODO: check if we can use annotations for field selectors)
```

Comment:
> TODO: check if we can use annotations for field selectors

Metadata fields are not allowed to be used, as per current implementation:
```go
if path.Root().String() == "metadata" {
	allErrs = append(allErrs, field.Invalid(indexFldPath.Child("jsonPath"), selectableField.JSONPath, "must not point to fields in metadata"))
}
```

See https://github.com/kubernetes/kubernetes/blob/v1.30.3/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L812-L814 for details.

I wonder if we can re-use the same validation rules in codegen. Thoughts?

Reply:
I agree that we could/should use the same validation rules in codegen to avoid surprises. I'll have that in a future PR, rather than adding to this one.